llvm/lib/Target/BPF/BTFDebug.cpp

```diff
@@ -976,11 +976,21 @@ void BTFDebug::visitMapDefType(const DIType *Ty, uint32_t &TypeId) {
   if (Tag != dwarf::DW_TAG_structure_type || CTy->isForwardDecl())
     return;
 
-  // Visit all struct members to ensure pointee type is visited
+  // Visit all struct members to ensure their types are visited.
   const DINodeArray Elements = CTy->getElements();
+  // In Aya/Rust, BPF maps are wrapped into nested types. Such wrapper types
+  // always have only one element. When encountering them, we need to visit its
+  // (only) element with an assumption that it might contain either an another
+  // wrapper, or an actual map struct.
+  const bool IsAWrapperType = Elements.size() == 1;
   for (const auto *Element : Elements) {
     const auto *MemberType = cast<DIDerivedType>(Element);
-    visitTypeEntry(MemberType->getBaseType());
+    if (IsAWrapperType) {
```

Comment:
doing this check inside the loop is a bit odd, since it will only ever happen if the loop runs for exactly one iteration. at the very least I think this should move outside the loop body, but also I wonder if there's a more sensible way to do this. What if we decide to have more than one element in the wrapper struct in the future?